### PR TITLE
Dave.151.update.deps.to.2.0.0.pre

### PR DESCRIPTION
**Pull request by davecaos**
This PR depends that Inaka/lasse#44 it will previously merged

### CODE REVIEW

File: test/http_server/http_base_handler.erl [hunk 1]
```diff
@@ -9,22 +8,20 @@
...
 
 content_types_provided(Req, State) ->
-  {Method, Req1} = cowboy_req:method(Req),
+  Method = cowboy_req:method(Req),
   Handler = case Method of
               <<"HEAD">> -> handle_head;
               _ -> handle_get
             end,
-  {[{<<"text/plain">>, Handler}], Req1, State}.
+  ct:pal("Method ~p , Handler ~p", [Method,Handler ]),
```
Comment (gadgetci) [inline test/http_server/http_base_handler.erl]: According to **Elvis**:
> Missing space right "," on line 23

Comment (gadgetci) [inline test/http_server/http_base_handler.erl]: According to **Elvis**:
> Remove the debug call to ct:pal/2 on line 23.

File: test/http_server/http_basic_auth_handler.erl [hunk 2]
```diff
@@ -21,10 +20,13 @@ allowed_methods(Req, State) ->
 
 is_authorized(Req, State) ->
     case cowboy_req:parse_header(<<"authorization">>, Req) of
-        {ok, {<<"basic">>, {<<"user">>, <<"pass">>}}, Req2} ->
-            {true, Req2, State};
-        {_, _, Req2} ->
-            {{false, <<>>}, Req2, State}
+        {basic, <<"user">>, <<"pass">>} ->
+            {true, Req, State};
+        {_, _, Req} ->
```
Comment (elbrujohalcon) [inline test/http_server/http_basic_auth_handler.erl]: Does cowboy return the header here?

File: test/http_server/http_sse_handler.erl [hunk 3]
```diff
@@ -4,14 +4,14 @@
...
-         init/3,
+         init/2,
          handle_notify/2,
          handle_info/2,
          handle_error/3,
          terminate/3
         ]).
 
-init(_InitArgs, _LastEventId, Req) ->
+init(_LastEventId, Req) ->
```
Comment (elbrujohalcon) [inline test/http_server/http_sse_handler.erl]: This is a `lasse_handler`, shouldn't it still receive those 3 args?

File: test/http_server/http_binary_handler.erl [hunk 4]
```diff
@@ -1,37 +1,37 @@
...
-    {<<"GET">>, Req1} ->
+    <<"GET">>  ->
       Headers = [{<<"content-type">>, <<"text/event-stream">>},
                  {<<"cache-control">>, <<"no-cache">>}],
-      {ok, Req2} = cowboy_req:chunked_reply(200, Headers, Req1),
+      Req1 = cowboy_req:chunked_reply(200, Headers, Req),
       shotgun_test_utils:auto_send(count),
-      {loop, Req2, 1};
-    {_, _} ->
+      {cowboy_loop, Req1, 1};
```
Comment (gadgetci) [inline test/http_server/http_binary_handler.erl]: According to **Elvis**:
> Line 17 has 1 trailing whitespace characters.

File: test/http_server/http_sse_handler.erl (hunk 3 above)
Comment (davecaos) [inline test/http_server/http_sse_handler.erl]: http_sse_handler should respect the contract init/2 made en in cowboy_loop behaviour in lasse_handler
https://github.com/inaka/lasse/compare/davecaos.cowboy2x?expand=1#diff-d3f14668f1b3c9cb057561392688186cR4

https://github.com/ninenines/cowboy/blob/2.0.0-pre.3/src/cowboy_loop.erl#L30

Comment (elbrujohalcon) [inline test/http_server/http_sse_handler.erl]: @davecaos that's not true.

File: Makefile [hunk 5]
```diff
@@ -2,17 +2,17 @@ PROJECT = shotgun
...
-DEPS       = cowlib gun
+DEPS       = gun
 BUILD_DEPS = inaka_mk hexer_mk
 TEST_DEPS  = katana_test cowboy mixer lasse
 SHELL_DEPS = sync
 
-dep_cowlib      = hex 1.0.2
-dep_gun         = hex 1.0.0-pre.1
+
+dep_gun         = git https://github.com/ninenines/gun.git     d88f3675d
```
Comment (elbrujohalcon) [inline Makefile]: This is a main dep (i.e. it's in `DEPS`). It should be fetched from hex

File: test/shotgun_async_SUITE.erl [hunk 6]
```diff
@@ -95,8 +95,7 @@ get_binary(Config) ->
 
     timer:sleep(500),
 
-    [Chunk1, Chunk2, Fin] = shotgun:events(Conn),
-
+    [Chunk1, Chunk2, Fin]  = shotgun:events(Conn),
```
Comment (elbrujohalcon) [inline test/shotgun_async_SUITE.erl]: extra space not needed

**Pull request merged**